test/pylib/util.py

```diff
@@ -281,8 +281,8 @@ def get_configured_modes() -> list[str]:
                             out, count=1, flags=re.DOTALL).split('\n')[-1].split(' ')
 
 
-def get_modes_to_run(session) -> list[str]:
-    modes = session.config.getoption('modes')
+def get_modes_to_run(config) -> list[str]:
```

Comment:
This changes get_modes_to_run to work on config, rather than session (sincerely, your Captain Obvious). It would be nice to have this change in separate patch